runtime/gc_realtime/RealtimeMarkingScheme.cpp

```diff
@@ -215,3 +209,98 @@ MM_RealtimeMarkingScheme::incrementalCompleteScan(MM_EnvironmentRealtime *env, u
 	}
 }
 
+uintptr_t
+MM_RealtimeMarkingScheme::scanObject(MM_EnvironmentRealtime *env, uintptr_t item, GC_ObjectScannerState *objectScannerState)
+{
+	uintptr_t scannedPointers = 0;
+	uintptr_t sizeToDo = UDATA_MAX;
+	GC_ObjectScanner *objectScanner = nullptr;
+	if (IS_ITEM_ARRAYLET((uintptr_t)objectPtr)) {
```

Comment:
I put the IS_ITEM_ARRAYLET check into scanObject instead of incrementalCompleteScan because that's where the created scanner is used. Creating the scanner in incrementalScan() and trying to use base scanObject() maybe doesn't seem right?